conda/base/context.py

```diff
@@ -609,6 +620,20 @@ def aggressive_update_packages(self):
         from ..models.match_spec import MatchSpec
         return tuple(MatchSpec(s) for s in self._aggressive_update_packages)
 
+    @memoizedproperty
+    def prerelease_allowed_specs(self):
+        from ..models.match_spec import MatchSpec
+        return tuple(MatchSpec(s) for s in context.prerelease_allowed_packages)
+
+    def is_prerelease(self, rec):
+        """True if PackageRecord or MatchSpec specifies a prerelease
+
+        Will return false if package is a prerelease but matches a spec in
+        prerelease_allowed_packages.
+        """
+        return (rec.is_prerelease and
+                not any(spec.match(rec) for spec in self.prerelease_allowed_specs))
```

Comment:
I am not sure I like this method here. I know you want to add a context-aware method but to me it feels foreign to have this kind of version-specific logic here.

Maybe we can remove it from here and leave the VersionOrder and MatchSpec implementations only, as methods (not properties) that take `prerelease_allowed_specs` as an optional kwarg, defaulting to None, which is replaced by the context option at call-time. Is that reasonable?

Reply:
Yes, I was a bit reluctant to put this here, but I was trying to avoid code duplication. I will consider that.